src/theory/strings/regexp_operation.h

```diff
@@ -58,7 +73,8 @@ class RegExpOpr {
   std::map<PairNodeStr, Node> d_dv_cache;
   std::map<PairNodeStr, std::pair<Node, int> > d_deriv_cache;
   std::map<Node, std::pair<Node, int> > d_compl_cache;
-  std::map<Node, bool> d_cstre_cache;
+  /** cache mapping regular expressions to whether they contain constants */
+  std::unordered_map<Node, RegExpConstType, NodeHashFunction> d_constCache;
```

Comment:
This is very nit-picky but I think a better name would be `d_regExpTypeCache`

Reply:
This would actually be `d_regExpConstTypeCache` though?  I prefer the shorter name; the other words there are not very unique to this cache.

Reply:
Hm, I just thought that `d_constCache` is very generic whereas `d_regExpTypeCache` is at least saying that we cache the type of regexps that we've seen. Not very critical though.